lib/libzfs/libzfs_sendrecv.c

```diff
@@ -941,28 +942,39 @@ send_iterate_fs(zfs_handle_t *zhp, void *arg)
 	}
 
 	if (nv != NULL)
-		VERIFY(0 == nvlist_add_nvlist(nvfs, "props", nv));
+		fnvlist_add_nvlist(nvfs, "props", nv);
 
 	/* iterate over snaps, and set sd->parent_fromsnap_guid */
 	sd->parent_fromsnap_guid = 0;
-	VERIFY(0 == nvlist_alloc(&sd->parent_snaps, NV_UNIQUE_NAME, 0));
-	VERIFY(0 == nvlist_alloc(&sd->snapprops, NV_UNIQUE_NAME, 0));
+	sd->parent_snaps = fnvlist_alloc();
+	sd->snapprops = fnvlist_alloc();
+	if (!sd->replicate && fromsnap_txg != 0) {
+		range_nvl = fnvlist_alloc();
+		fnvlist_add_uint64(range_nvl, SNAP_ITER_SKIP_TO_TXG,
+		    fromsnap_txg_save);
+	}
+	if (!sd->replicate && tosnap_txg != 0) {
+		if (range_nvl == NULL)
+			range_nvl = fnvlist_alloc();
+		fnvlist_add_uint64(range_nvl, SNAP_ITER_SKIP_AFTER_TXG,
+		    sd->tosnap_txg);
```

Comment:
It looks like if `tosnap_txg` is nonzero then `sd->tosnap_txg` will be the same, but I think this assumption make the code harder to understand.  I think we should be using the local `tosnap_txg` here.